components/remote_settings/src/client.rs

```diff
@@ -80,6 +107,52 @@ impl<C: ApiClient> RemoteSettingsClient<C> {
         // Try to retrieve and filter cached records first
         let cached_records = inner.storage.get_records(&collection_url)?;
 
+        // Check if we should load packaged data
+        let should_load_packaged = match &cached_records {
+            None => true,                                // No local data, try packaged data
+            Some(records) if records.is_empty() => true, // Empty local data, try packaged data
+            _ => false,
+        };
+
+        if should_load_packaged {
+            // Only load packaged data when pointing to PROD server
+            if inner.api_client.is_prod_server() {
+                if let Some(packaged_records) = self.load_packaged_data()? {
+                    // Store packaged data in the database
+                    inner
+                        .storage
+                        .set_records(&collection_url, &packaged_records)?;
+                    let filtered_records = self.filter_records(packaged_records);
+                    return Ok(Some(filtered_records));
+                }
+            }
+        } else {
+            // If we have local data, check if packaged data is newer
+            if inner.api_client.is_prod_server() {
+                if let Some(packaged_records) = self.load_packaged_data()? {
+                    if let Some(cached) = &cached_records {
+                        // Compare timestamps
+                        let cached_timestamp =
+                            cached.iter().map(|r| r.last_modified).max().unwrap_or(0);
+                        let packaged_timestamp = packaged_records
+                            .iter()
+                            .map(|r| r.last_modified)
+                            .max()
+                            .unwrap_or(0);
```

Comment:
Don't we store the collection timestamp? 

Using the max of records timestamps won't work for deletions, because tombstones are not stored.